api/v1beta1/cinder_webhook.go

```diff
@@ -374,3 +461,50 @@ func (spec *CinderSpec) ValidateCinderTopology(basePath *field.Path, namespace s
 	}
 	return allErrs
 }
+
+// TODO: Remove this function when refactoring CinderSpec to include CinderSpecCore
+func (spec *CinderSpec) ValidateCinderBackup(basePath *field.Path) ([]string, field.ErrorList) {
+	var allErrs field.ErrorList
+	var allWarns []string
+
+	// Check if the deprecated field is configured
+	isDeprecatedUsed := spec.CinderBackup.Replicas != nil && *spec.CinderBackup.Replicas > 0
+	// Check if CinderBackup list is configured
+	isCinderBackupListUsed := spec.CinderBackups != nil && len(*spec.CinderBackups) > 0
+
+	// Fail if both cinderBackup (the deprecated field) and cinderBackups (the new list)
+	// are used together
+	if isDeprecatedUsed && isCinderBackupListUsed {
+		errMsg := "Usage of the deprecated 'cinderBackup' is forbidden when 'cinderBackups' is defined."
+		allErrs = append(allErrs, field.Invalid(basePath, "cinderBackup", errMsg))
+	}
+	// Append a warning depending on the cinderBackup field usage
+	if isDeprecatedUsed {
+		warningMsg := "The 'cinderBackup' field is deprecated and will be removed in a future release. Please migrate to 'cinderBackups'."
+		allWarns = append(allWarns, warningMsg)
+	}
+	return allWarns, allErrs
+}
+
+func (spec *CinderSpecCore) ValidateCinderBackup(basePath *field.Path)([]string, field.ErrorList) {
+	var allErrs field.ErrorList
+	var allWarns []string
+
+	// Check if the deprecated field is configured
+	isDeprecatedUsed := spec.CinderBackup.Replicas != nil && *spec.CinderBackup.Replicas > 0
+	// Check if CinderBackup list is configured
+	isCinderBackupListUsed := spec.CinderBackups != nil && len(*spec.CinderBackups) > 0
+
+	// Fail if both cinderBackup (the deprecated field) and cinderBackups (the new list)
+	// are used together
+	if isDeprecatedUsed && isCinderBackupListUsed {
+		errMsg := "Usage of the deprecated 'cinderBackup' is forbidden when the new 'cinderBackups' API is used."
+		allErrs = append(allErrs, field.Invalid(basePath, "cinderBackup", errMsg))
+	}
+	// Append a warning depending on the cinderBackup field usage
+	if isDeprecatedUsed {
+		warningMsg := "The 'cinderBackup' field is deprecated and will be removed in a future release. Please migrate to 'cinderBackups'."
+		allWarns = append(allWarns, warningMsg)
+	}
+	return allWarns, allErrs
+}
```

Comment:
Even though this isn't DRY, seems fine for now if we're going to soon refactor the API and clean this up anyhow.